tests/integration/sync_test.go

```diff
@@ -25,82 +26,296 @@ func getKongAddress() string {
 	return "http://localhost:8001"
 }
 
-// cleanUpEnv removes all existing entities from Kong.
-func cleanUpEnv(t *testing.T) error {
-	deckCmd.SetArgs([]string{"reset", "--force"})
-	if err := deckCmd.Execute(); err != nil {
-		t.Errorf(err.Error())
-	}
-	return nil
-}
-
 func getTestClient() (*kong.Client, error) {
 	return utils.GetKongClient(utils.KongClientConfig{
 		Address: getKongAddress(),
 	})
 }
 
-func testKongState(
-	t *testing.T,
-	client *kong.Client,
-	expectedServices []*kong.Service,
-) {
-	ctx := context.Background()
+func sortSlices(x, y interface{}) bool {
+	var xName, yName string
+	switch xEntity := x.(type) {
+	case *kong.Service:
+		yEntity := y.(*kong.Service)
+		xName = *xEntity.Name
+		yName = *yEntity.Name
+	case *kong.Route:
+		yEntity := y.(*kong.Route)
+		xName = *xEntity.Name
+		yName = *yEntity.Name
+	case *kong.Plugin:
+		yEntity := y.(*kong.Plugin)
+		xName = *xEntity.Name
+		yName = *yEntity.Name
+	}
+	return xName < yName
+}
+
+func testKongState(t *testing.T, client *kong.Client, expectedState utils.KongRawState) {
 	// Get entities from Kong
-	services, err := dump.GetAllServices(ctx, client, []string{})
+	ctx := context.Background()
+	kongState, err := dump.Get(ctx, client, dump.Config{})
 	if err != nil {
 		t.Errorf(err.Error())
 	}
+
 	opt := []cmp.Option{
 		cmpopts.IgnoreFields(kong.Service{}, "ID", "CreatedAt", "UpdatedAt"),
+		cmpopts.IgnoreFields(kong.Route{}, "ID", "CreatedAt", "UpdatedAt",
+			"Service", "RequestBuffering", "ResponseBuffering", "PathHandling"),
+		cmpopts.IgnoreFields(kong.Plugin{}, "ID", "CreatedAt"),
+		cmpopts.SortSlices(sortSlices),
+		cmpopts.EquateEmpty(),
 	}
-	if diff := cmp.Diff(services, expectedServices, opt...); diff != "" {
+
+	if diff := cmp.Diff(kongState, &expectedState, opt...); diff != "" {
 		t.Errorf(diff)
 	}
 }
 
+func reset(t *testing.T) {
+	deckCmd.SetArgs([]string{"reset", "--force"})
+	if err := deckCmd.Execute(); err != nil {
+		t.Errorf(err.Error())
+	}
+}
+
+func setup(t *testing.T, kongFile string) func(t *testing.T) {
+	if kongFile != "" {
+		sync(t, kongFile)
+	}
+	return func(t *testing.T) {
+		reset(t)
+	}
+}
+
+func sync(t *testing.T, kongFile string) {
+	// directly set flag value due to cobra keeping state across test runs
+	cmd.SyncCmdKongStateFile = []string{kongFile}
+	deckCmd.SetArgs([]string{"sync"})
+	err := deckCmd.Execute()
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+}
+
+func getKongVersionFromEnv() string {
+	return strings.Split(os.Getenv("KONG_IMAGE"), ":")[1]
+}
+
 func Test_Sync(t *testing.T) {
 	// setup stage
 	client, err := getTestClient()
 	if err != nil {
 		t.Errorf(err.Error())
 	}
 
+	kongVersion := getKongVersionFromEnv()
+
 	tests := []struct {
-		name             string
-		kongFile         string
-		expectedServices []*kong.Service
+		name            string
+		kongFile        string
+		initialKongFile string
```

Comment:
Question: why two files? Where these variables are used, both are exactly the same, what is it the purpose here?

Reply:
The intent was to use this in order to test `sync` with an existing state, rather than only test results from a clean one.

Reply:
I'd recommend having separate TestFuncs for these. Fresh vs Diffs.